frontend/src/types/api.ts

```diff
@@ -1,5 +1,14 @@
+import { z } from "zod";
+
 import type { SortDirection } from "./utils";
 
+// Custom regex since Zod .datetime() requires either an offset or Z,
+// while our API dates does not
+const ISO_8601_REGEX =
+  /^\d{4}-\d\d-\d\dT\d\d:\d\d:\d\d(\.\d+)?(([+-]\d\d:\d\d)|Z)?$/i;
+export const apiDateSchema = z.string().regex(ISO_8601_REGEX);
```

Comment:
Would a custom refinement that uses native `try { new Date(...) } ...` maybe avoid the need for the regex? No particularly strong preference, but it'd prevent any issues where the regex doesn't exactly align with how the native API works

Reply:
I aimed for something slightly improved from the `date: string;` + inline comment we had before for this PR, we should discuss the future of API dates with @tw4l in Discord, maybe it'd be better to just add `Z`/an offset in the API?

Reply:
Yeah, that's also an option! My concern with a custom regex is that the [Date Time String Format defined in the specification](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-date-time-string-format) is a simplification of ISO8601, so matching against an ISO8601 regex _might_ cause issues. That said, this is a significant improvement over what we previously had either way!

A refinement that uses `Date` might look like this:
```ts
z.string().refine((v) => !isNaN(Date.parse(v)), { message: "String must be a valid date" })
```

Reply:
Ah I see what you mean now, yeah the refinement is much nicer than regex!